translib/transformer/test/transformer_test.go

```diff
@@ -0,0 +1,299 @@
+
+	if isGetCase {
+		isWriteDisabled = true
+	} else {
+		isWriteDisabled = false
+	}
+
+	//Create Counter DB connection
+	dbs[db.CountersDB], err = db.NewDB(getDBOptions(db.CountersDB, isWriteDisabled))
+
+	if err != nil {
+		closeAllDbs(dbs[:])
+		return dbs, err
+	}
+        
+        //Create Config DB connection
+	dbs[db.ConfigDB], err = db.NewDB(getDBOptions(db.ConfigDB, isWriteDisabled))
+
+	if err != nil {
+		closeAllDbs(dbs[:])
+		return dbs, err
+	}
+
+	return dbs, err
+}
+
+// Closes the dbs, and nils out the arr.
+func closeAllDbs(dbs []*db.DB) {
+	for dbsi, d := range dbs {
+		if d != nil {
+			d.DeleteDB()
+			dbs[dbsi] = nil
+		}
+	}
+}
+
+func TestMain(t *testing.M) {
+	fmt.Println("----- Setting up transformer tests -----")
+	if err := setup(); err != nil {
+		fmt.Fprintf(os.Stderr, "Error setting up transformer testing state: %v.\n", err)
+		os.Exit(1)
+	}
+	t.Run()
+	teardown()
+	os.Exit(0)
+}
+
+
+func initDbConfig() error {
+	dbConfigFile := "/run/redis/sonic-db/database_config.json"
+	if path, ok := os.LookupEnv("DB_CONFIG_PATH"); ok {
+		dbConfigFile = path
+	}
+
+	fmt.Println("dbConfigFile =", dbConfigFile)
+	dbConfigJson, err := ioutil.ReadFile(dbConfigFile)
+	if err == nil {
+		err = json.Unmarshal(dbConfigJson, &dbConfig)
+	}
+
+	return err
+}
+
+func clearDb() {
+
+	tblList := []string{
+		"TEST_SENSOR_GROUP",
+		"TEST_SENSOR_A_TABLE",
+		"TEST_SENSOR_B_TABLE",
+		"TEST_SET_TABLE",
+		"TEST_SENSOR_MODE_TABLE",
+	}
+
+	for _, tbl := range tblList {
+		_, err := exec.Command("/bin/sh", "-c",
+			"sonic-db-cli CONFIG_DB del `sonic-db-cli CONFIG_DB keys '"+
+				tbl+"|*' | cut -d ' ' -f 2`").Output()
+
+		if err != nil {
+			fmt.Println(err.Error())
+		}
+	}
+}
+
+/* Prepares the database in Redis Server. */
+func prepareDb() {
+	rclient = getConfigDbClient()
+	if rclient == nil {
+		fmt.Printf("error in getConfigDbClient")
+		return
+	}
+
+	rclientDBNum = make(map[db.DBNum]*redis.Client)
+	/*Add redis client for specific DB as and how needed*/
+	rclientDBNum[db.CountersDB] = getDbClient(int(db.CountersDB))
+	if rclientDBNum[db.CountersDB] == nil {
+		fmt.Printf("error in getDbClient(int(db.CountersDB)")
+		return
+	}
+	rclientDBNum[db.ConfigDB] = getDbClient(int(db.ConfigDB))
+	if rclientDBNum[db.ConfigDB] == nil {
+		fmt.Printf("error in getDbClient(int(db.ConfigDB)")
+		return
+	}
+
+}
+
+// setups state each of the tests uses
+func setup() error {
+	fmt.Println("----- Performing setup -----")
+	var err error
+	if ygSchema, err = ocbinds.Schema(); err != nil {
+		panic("Error in getting the schema: " + err.Error())
+		return err
+        }
+
+	if err := initDbConfig(); err != nil {
+		return err
+	}
+
+	//Clear all tables which are used for testing
+	clearDb()
+
+	/* Prepare the Redis database. */
+	prepareDb()
+
+	return nil
+}
+
+func teardown() error {
+	if rclient == nil {
+		return nil
+	}
+
+	fmt.Println("----- Performing teardown -----")
+	clearDb()
+	rclient.Close()
+	rclient.FlushDB()
```

Comment:
`FlushDB()` deletes all keys from the selected DB. Hence the previous `clearDb()` call is totally redundant. Please keep only one of them.
IMO `FlushDB()` is dangerous as it will delete already existing entries, like the PORT, DEVICE_METADATA etc. Better to delete the keys created by our test code only.

Reply:
Addressed as suggested

Reply:
`FlushDB()` call still present

Reply:
Done. Added the corrected file missing in previous commit.